pyfifotax/report_data.py

```diff
@@ -0,0 +1,436 @@
+                    + event.withheld_shares.total_buy_value(),
+                )
+                bought_shares = awv.AWVEntryZ10RSUDeposit(
+                    date=event.date,
+                    symbol=event.symbol,
+                    currency=event.received_shares.currency,
+                    quantity=event.received_shares.quantity,
+                    value=event.received_shares.total_buy_value()
+                    + event.withheld_shares.total_buy_value(),
+                )
+                withheld_shares = awv.AWVEntryZ10RSUTaxWithholding(
+                    date=event.date,
+                    symbol=event.symbol,
+                    currency=event.received_shares.currency,
+                    quantity=event.withheld_shares.quantity,
+                    value=event.received_shares.total_buy_value(),
+                )
+                self.awv_z4_events.append(bonus)
+                self.awv_z10_events.append(bought_shares)
+                if event.withheld_shares.quantity > 0:
+                    self.awv_z10_events.append(withheld_shares)
+
+            elif isinstance(event, ESPPEvent):
+                self.held_shares[event.symbol].push(event.received_shares)
+
+                bonus = awv.AWVEntryZ4ESPPBonus(
+                    symbol=event.symbol,
+                    currency=event.currency,
+                    date=event.date,
+                    value=event.bonus,
+                )
+
+                bought_shares = awv.AWVEntryZ10ESPPDeposit(
+                    date=event.date,
+                    symbol=event.symbol,
+                    currency=event.currency,
+                    quantity=event.received_shares.quantity,
+                    value=event.received_shares.total_buy_value(),
+                )
+
+                self.awv_z4_events.append(bonus)
+                self.awv_z10_events.append(bought_shares)
+
+            elif isinstance(event, DividendEvent):
+                if event.currency != self.domestic_currency:
+                    self.held_forex[event.currency].push(event.received_net_dividend)
+
+                self.misc["Tax Withholding"].append(event.withheld_tax)
+                self.misc["Dividend Payments"].append(event.received_dividend)
+
+                # dividends should be small enough to not trigger AWV reportings
+
+            elif isinstance(event, BuyEvent):
+                if event.currency != self.domestic_currency:
+                    # if not enough money, pop on FOREX Queue will fail
+                    tmp = self.held_forex[event.currency].pop(
+                        event.cost_of_shares,
+                        to_decimal(1),
+                        event.date,
+                    )
+                    self.sold_forex[event.currency].extend(tmp)
+
+                self.held_shares[event.symbol].push(event.received_shares)
+                self.misc["Fees"].append(event.paid_fees)
```

Comment:
Regression: 0 fees are appearing in the report again. In the original code, `report_data.py:173` you can find a function which takes care of this (and has a warning for negative fees)

Reply:
Good catch, I'll add this again.